File: statsig-java/src/main/java/com/statsig/ParameterStore.java
package com.statsig;

import com.alibaba.fastjson2.annotation.JSONCreator;
import com.alibaba.fastjson2.annotation.JSONField;
import java.util.Map;

public class ParameterStore {
  public String name;
  public EvaluationDetails evaluationDetails;

  private Statsig statsigInstance;
  private StatsigUser user;

  @JSONCreator
  ParameterStore(
      @JSONField(name = "name") String name,
      @JSONField(name = "details") EvaluationDetails evaluationDetails) {
    this.name = name;
    this.evaluationDetails = evaluationDetails;
  }

  public String getName() {
    return name;
  }

  public EvaluationDetails getEvaluationDetails() {
    return evaluationDetails;
  }

  void setStatsigInstance(Statsig statsigInstance) {
    this.statsigInstance = statsigInstance;
  }

  void setUser(StatsigUser user) {
    this.user = user;
  }

  public String getString(String parameterName, String defaultValue) {
    String value =
        statsigInstance.getStringFromParameterStore(user, name, parameterName, defaultValue);
    if (value == null) {
      return defaultValue;
    }
    return value;
  }

  public boolean getBoolean(String parameterName, boolean defaultValue) {
    return statsigInstance.getBooleanFromParameterStore(user, name, parameterName, defaultValue);
  }

  public double getDouble(String parameterName, double defaultValue) {
    return statsigInstance.getDoubleFromParameterStore(user, name, parameterName, defaultValue);
  }

  public int getInt(String parameterName, int defaultValue) {
    return statsigInstance.getIntFromParameterStore(user, name, parameterName, defaultValue);
  }

  public long getLong(String parameterName, long defaultValue) {
    return statsigInstance.getLongFromParameterStore(user, name, parameterName, defaultValue);
  }

  public Object[] getArray(String parameterName, Object[] defaultValue) {
    return statsigInstance.getArrayFromParameterStore(user, name, parameterName, defaultValue);
  }

  public Map<String, Object> getMap(String parameterName, Map<String, Object> defaultValue) {
    return statsigInstance.getMapFromParameterStore(user, name, parameterName, defaultValue);
  }
}

File: statsig-go/parameter_store.go
package statsig

type ParameterStore struct {
	Name                   string
	Details                map[string]interface{} `json:"details"`
	Options                ParameterStoreOptions  `json:"options"`
	statsigInstance        Statsig
	user                   StatsigUser
	disableExposureLogging bool
}

func (ps *ParameterStore) setStatsigUser(u *StatsigUser) {
	ps.user = *u
}

func (ps *ParameterStore) setStatsigInstance(s *Statsig) {
	ps.statsigInstance = *s
}

func (ps *ParameterStore) setDisableExposureLogging(disableExposureLogging bool) {
	ps.disableExposureLogging = disableExposureLogging
}

func (ps *ParameterStore) GetString(parameterName string, defaultValue string) string {
	return ps.statsigInstance.GetStringFromParameterStore(ps.user, ps.Name, parameterName, defaultValue, &ps.Options)
}

func (ps *ParameterStore) GetBoolean(parameterName string, defaultValue bool) bool {
	return ps.statsigInstance.GetBooleanFromParameterStore(ps.user, ps.Name, parameterName, defaultValue, &ps.Options)
}

func (ps *ParameterStore) GetFloat64(parameterName string, defaultValue float64) float64 {
	return ps.statsigInstance.GetFloat64FromParameterStore(ps.user, ps.Name, parameterName, defaultValue, &ps.Options)
}

func (ps *ParameterStore) GetInt(parameterName string, defaultValue int) int {
	return ps.statsigInstance.GetIntFromParameterStore(ps.user, ps.Name, parameterName, defaultValue, &ps.Options)
}

func (ps *ParameterStore) GetInt64(parameterName string, defaultValue int64) int64 {
	return ps.statsigInstance.GetInt64FromParameterStore(ps.user, ps.Name, parameterName, defaultValue, &ps.Options)
}

func (ps *ParameterStore) GetInterface(parameterName string, defaultValue []interface{}) []interface{} {
	return ps.statsigInstance.GetInterfaceFromParameterStore(ps.user, ps.Name, parameterName, defaultValue, &ps.Options)
}

func (ps *ParameterStore) GetMap(parameterName string, defaultValue map[string]interface{}) map[string]interface{} {
	return ps.statsigInstance.GetMapFromParameterStore(ps.user, ps.Name, parameterName, defaultValue, &ps.Options)
}
